stripped/colabs/agregacje_kilkuletnie_stripped.py

```diff
@@ -0,0 +1,669 @@
+from pyspark.sql import SparkSession
+from pyspark.sql.types import IntegerType, FloatType, StringType, StructType
+from pyspark.sql import Window
+import pyspark
+from sklearn.ensemble import RandomForestClassifier
+import lightgbm as lgb
+
+"""# Załadowanie zbioru danych"""
+
+from google.colab import drive
+drive.mount("/content/drive")
+
+!git clone https://github.com/PiotrMaciejKowalski/BigData2024Project.git
+
+!chmod 755 /content/BigData2024Project/src/setup.sh
+!/content/BigData2024Project/src/setup.sh
+
+import sys
+sys.path.append('/content/BigData2024Project/src')
+
+from start_spark import initialize_spark
+initialize_spark()
+
+from pyspark.sql import SparkSession
+from big_mess.loaders import default_loader
+from pyspark.sql import SparkSession, DataFrame as SparkDataFrame, functions as F
+
+spark = SparkSession.builder\
+        .master("local")\
+        .appName("Colab")\
+        .config('spark.ui.port', '4050')\
+        .getOrCreate()
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# nasa = default_loader(spark)
+# nasa.show(5)
+
+nasa.createOrReplaceTempView("nasa")
+
+"""# Średnie wieloletnie dla poszczególnych miesięcy"""
+
+def overYearMonthStats(columns: list, n: int):
+    """
+    Funkcja licząca średnie dla wybranych zmiennych dla danych współrzędnych geograficznych
+    dla każdego z 12 miesięcy od roku w rozważanym pomiarze do n lat wstecz (tj. gdyby n=3, to
+    średnia zostałaby wyliczona z rozważanego roku oraz 3 lat przed nim, czyli w praktyce mielibyśmy
+    średnią z łącznie 4 lat).
+
+    :param columns: lista nazw rozważanych kolumn ze zbioru danych
+    :param n: liczba lat wstecz, z których chcemy wyliczyć średnią
+    """
+
+    if not columns:
+      raise ValueError("Podana musi zostać co najmniej jedna zmienna")
+
+    windowSpec = Window.partitionBy("lon", "lat", "Month").orderBy("Year").rowsBetween(-n, 0)
+
+    nasa_window = nasa
+    for column in columns:
+        average_column_name = "monthly_avg_" + str(n) + "years_" + column
+        nasa_window = nasa_window.withColumn(average_column_name, F.avg(F.col(column)).over(windowSpec))
+
+    return nasa_window
+
+"""Zastosujmy teraz powyższą funkcję na naszym datasecie, dla każdej z rozważanych kolumn pomiarowych, przyjmując n=3:"""
+
+columns_to_avg = ["Rainf", "Evap", "AvgSurfT", "Albedo", "SoilT_40_100cm", "GVEG", "PotEvap", "RootMoist", "SoilM_100_200cm"]
+
+result_df_3 = overYearMonthStats(columns_to_avg, n=3)
+result_df_3.show(15)
+
+"""Wyodrębnijmy teraz kilka lat do oddzielnych dataframe'ów i zapiszmy je:"""
+
+# 2023
+nasa_2023_monthly_avg_3 = result_df_3.filter(f'Year = {2023}')
+pd_2023_monthly_avg_3 = nasa_2023_monthly_avg_3.toPandas()
+
+pd_2023_monthly_avg_3.to_csv('/content/drive/MyDrive/BigMess/NASA/monthly_avg/3years/NASA_monthly_avg_2023.csv')
+
+# 2022
+nasa_2022_monthly_avg_3 = result_df_3.filter(f'Year = {2022}')
+pd_2022_monthly_avg_3 = nasa_2022_monthly_avg_3.toPandas()
```

Comment:
Kod dalej jest strasznie powtarzalny ... Niech już będzie, ale to jest zła praktyka.

Reply:
Do momentu przejścia do treningów zajęłam się już kwestią powtarzalności. Powinno być lepiej, a kod jest sporo krótszy